### PR TITLE
As requested, `*` now vectorises if there's a function

### LINKED ISSUE #748: `*` doesn't vectorise on functions.
Don't ask why I needed this.

Seriously. Don't.

Also we should *probably* put that behaviour of `*` (and `Td`) in the docs.

### PR DESCRIPTION
**Pull request by Lyxal**
Closes #748 

Honestly idk why on earth you'd need this, but as you wish.

### CODE REVIEW

File: vyxal/helpers.py [hunk 1]
```diff
@@ -589,7 +589,9 @@ def gen():
 def primitive_type(item: Any) -> Union[str, type]:
     """Turns int/Rational/str into 'Scalar' and everything else
     into list"""
-    if type(item) in [int, sympy.Rational, str] or is_sympy(item):
+    if type(item) in [int, sympy.Rational, str, types.FunctionType] or is_sympy(
```
Comment (cgccuser) [inline vyxal/helpers.py:592]: You can also use `isinstance(item, (int, sympy.Rational, str, types.FunctionType))` for that `type ... in` check

File: vyxal/elements.py [hunk 2]
```diff
@@ -2418,10 +2418,14 @@ def multiply(lhs, rhs, ctx):
     ts = vy_type(lhs, rhs)
```
Comment (cgccuser) [inline vyxal/elements.py:2418]: You could set `simple=True` here and then just check for `list` below

**Review by cgccuser — APPROVED** [review]
Minor suggestions

**Review by Seggan — APPROVED** [review]
I understand none of this tbh, blind approval

**Pull request merged**